runtime/shared_common/CacheMap.cpp

```diff
@@ -7946,46 +7982,86 @@ SH_CacheMap::getPrereqCache(J9VMThread* currentThread, const char* cacheDir, SH_
-		I_8 layer = _sharedClassConfig->layer;
-		if (0 == layer) {
-			result = 0;
-			goto done;
-		}
+/**
+ *  Store the actual cache unique ID in the previous cache 
+ *
+ *	@param [in] currentThread The current JVM thread
+ *	@pamra [in] the cache directory
+ *	@param [in] createtime The cache create time which is stored in OSCache_header2.
+ *	@param [in] metadataBytes  The size of the metadata section of current oscache.
+ *	@param [in] classesBytes  The size of the classes section of current oscache.
+ *	@param [in] lineNumTabBytes  The size of the line number table section of current oscache.
+ *	@param [in] varTabBytes  The size of the  variable table section of current oscache.
+ *	@param [out] the Unique ID of a re-requiste cache.
+ *	@param [out] the length of the Unique ID string.
+ *
+ *	@return 1 This cache depends on a lower layer cache. The unique ID of the re-requisite cache is found in or stored to this cache as metadata.
+ *			0 This cache does not depend on a low layer cache.
+ *			CM_CACHE_CORRUPT if cache is corrupted.
+ *			CM_READ_CACHE_FAILED if failed to get the existing pre-requisite cache ID from metadata.
+ *			CM_CACHE_STORE_PREREQ_ID_FAILED if failed to store the pre-requisite cache ID
+ *
+ *
+ */
+IDATA
+SH_CacheMap::storeCacheUniqueID(J9VMThread* currentThread, const char* cacheDir, U_64 createtime, UDATA metadataBytes, UDATA classesBytes, UDATA lineNumTabBytes, UDATA varTabBytes, const char** prereqCacheID, UDATA* idLen)
+{
+	IDATA result = 0;
 
-		SH_OSCache::generateCacheUniqueID(currentThread, cacheDir, _cacheName, layer - 1, getCacheTypeFromRuntimeFlags(*_runtimeFlags), key, J9SHR_UNIQUE_CACHE_ID_BUFSIZE);
-		UDATA keylen = strlen(key);
+	if (UnitTest::CORRUPT_CACHE_TEST == UnitTest::unitTest) {
+		return 0;
+	}
 
-		J9UTF8* utfKeyStruct = (J9UTF8*)utfKeyPtr;
-		J9UTF8_SET_LENGTH(utfKeyStruct, (U_16)keylen);
-		memcpy((char*)J9UTF8_DATA(utfKeyStruct), key, keylen);
+	SH_ScopeManager* localSCM = getScopeManager(currentThread);
+	if (NULL == localSCM) {
+		Trc_SHR_CM_storeCacheUniqueID_Failed_To_Get_Manager(currentThread);
+		result = CM_CACHE_STORE_PREREQ_ID_FAILED;
+		return result;
+	}
 
-		tokenKey = addScopeToCache(currentThread, utfKeyStruct, TYPE_PREREQ_CACHE);
-		if (NULL == tokenKey) {
-			Trc_SHR_CM_getPrereqCache_Failed_To_Store_Prereq_UniqueID(currentThread, J9UTF8_LENGTH(tokenKey), (char*)J9UTF8_DATA(tokenKey));
-			result = CM_CACHE_STORE_PREREQ_ID_FAILED;
-			goto done;
-		}
-		result = 1;
-		*prereqCacheID = (const char*)J9UTF8_DATA(tokenKey);
-		*idLen = J9UTF8_LENGTH(tokenKey);
+	const J9UTF8* tokenKey = NULL;
+	char utfKey[J9SHR_UNIQUE_CACHE_ID_BUFSIZE];
+	char key[J9SHR_UNIQUE_CACHE_ID_BUFSIZE];
+	char* utfKeyPtr = (char*)&utfKey;
+
+	I_8 layer = _sharedClassConfig->layer;
+	if (0 == layer) {
+		result = 0;
+		return result;
 	}
 
-done:
+	Trc_SHR_CM_storeCacheUniqueID_generateCacheUniqueID_entry(currentThread, createtime, metadataBytes, classesBytes, lineNumTabBytes, varTabBytes);
```

Comment:
Entry trace point needs to be at the the start of the function. i.e line 8022 here.